lesson02/job1/main.py

```diff
@@ -0,0 +1,69 @@
+"""
+This file contains the controller that accepts command via HTTP
+and trigger business logic layer
+"""
+
+import os
+import sys
+from flask import Flask, request
+from flask import typing as flask_typing
+import api
+import storage
+
+
+AUTH_TOKEN = os.environ.get("API_AUTH_TOKEN")
+
+if not AUTH_TOKEN:
+    print("AUTH_TOKEN environment variable must be set")
+    sys.exit()
+
+
+app = Flask(__name__)
+
+
+@app.route('/', methods=['POST'])
+def main() -> flask_typing.ResponseReturnValue:
+    """
+    Controller that accepts command via HTTP and
+    trigger business logic layer
+
+    Proposed POST body in JSON:
+    {
+      "date: "2022-08-09",
+      "raw_dir": "/path/to/my_dir/raw/sales/2022-08-09"
+    }
+    """
+    input_data: dict = request.json
+    msg = ''
+    code = 0
+
+    if "date" in input_data and "raw_dir" in input_data:
+        #print(input_data, input_data['date'])
+
+        mylist = api.get_sales(input_data['date'])
+        #print(mylist[0])
+        #print(mylist[1])
+        code = int(mylist[1])
+
+        stored = storage.save_to_disk(
+                    mylist[0]['data'],
+                    input_data['raw_dir']+"/sales_"+input_data['date']
+        )
+
+        if mylist[1] == 200 and stored == 'ok':
```

Comment:
1. `stored == 'ok'` doesn't make sense because if there's a problem occurs with file savine, you'd got an exception and your code here wouldn't be reached in this place

2. `mylist[1] == 200` doesn't make a real sense as well. 

If I understood correctly, you're afraid to get no data from the API. If I'm right, just raising an exception is enough in data engineering area. If you want to be more precise, you can catch this exception here an form a correct resonce.